Code/Framework/AzCore/Platform/iOS/AzCore/NativeUI/NativeUISystemComponent_iOS.mm

```diff
@@ -10,16 +10,65 @@
 
 #import <UIKit/UIKit.h>
 
+#if defined(CARBONATED)
+#include <chrono>
+#endif
+
 namespace AZ
 {
     namespace NativeUI
     {
-#if defined(CARBONATED) // TODO : implement for platform
+#if defined(CARBONATED)
+        // TODO : implement for platform
         bool NativeUISystem::IsDisplayingBlockingDialog() const
         {
             return false;
         }
+    
+        template <
+            class result_t   = std::chrono::milliseconds,
+            class clock_t    = std::chrono::steady_clock,
+            class duration_t = std::chrono::milliseconds
+        >
+        auto TimeSince(std::chrono::time_point<clock_t, duration_t> const& start)
+        {
+            return std::chrono::duration_cast<result_t>(clock_t::now() - start);
+        }
+    
+        static void OnDeadlock(NSString* nsTitle, NSString* nsMessage)
+        {
+            // Save the popup data into a file and abort the application.
+            // The file will be located in the Documents dir and can be fetched to Mac as a part of the app container.
+            NSDate* date = [NSDate date];
+            NSDateFormatter* formatter = [[NSDateFormatter alloc] init];
+            NSTimeZone* destinationTimeZone = [NSTimeZone systemTimeZone];
+            formatter.timeZone = destinationTimeZone;
+            [formatter setDateStyle:NSDateFormatterLongStyle];
+            [formatter setDateFormat:@"MM-dd-yyyy hh.mma"];
+            NSString* dateString = [formatter stringFromDate:date];
+            
+            // Include the date in the file name to not overwrite previous artifacts.
+            NSString* fileName = [NSString stringWithFormat:@"BlockingDialogAborted %@.txt", dateString];
+            
+            NSArray* allPaths = NSSearchPathForDirectoriesInDomains(NSDocumentDirectory, NSUserDomainMask, YES);
+            NSString* documentsDirectory = [allPaths objectAtIndex:0];
+            NSString* pathForLog = [documentsDirectory stringByAppendingPathComponent:fileName];
+
+            // Redirect NSLog messages to the file.
+            freopen([pathForLog cStringUsingEncoding:NSASCIIStringEncoding], "a+", stderr);
+            
+            // These messages go to the file.
+            NSLog(@"BlockingDialog failed to display at %@", dateString);
+            
+            NSLog(@"Title: %@", nsTitle);
+            NSLog(@"Message: %@\n", nsMessage);
+            
+            NSLog(@"App terminated");
+            
+            abort();
```

Comment:
Maybe it could better to try to continue instead of unexpected abort.
Not the all asserts are fatal.
At least, if they cause the next crash, we will see in the log, that the assert occurred before the crash.

Reply:
(For example: usually we press "Ignore" or "Ignore All" instead "Break". Nobody press "Break".)